packages/recoil/hooks/Recoil_Hooks.js

```diff
@@ -46,6 +46,8 @@ const expectationViolation = require('recoil-shared/util/Recoil_expectationViola
 const gkx = require('recoil-shared/util/Recoil_gkx');
 const recoverableViolation = require('recoil-shared/util/Recoil_recoverableViolation');
 const useComponentName = require('recoil-shared/util/Recoil_useComponentName');
+const {isSSR} = require('recoil-shared/util/Recoil_Environment');
+const {isPromise} = require('recoil-shared/util/Recoil_isPromise');
```

Comment:
Hmm, this shouldn't work as-is since `Recoil_isPromise.js` exports the function directly.  Is this the version you tested with?

I updated the diff to fix this import and use `finally()` instead, though unfortunately the changes won't sync back to this PR.  I'll test with internal unit tests but we don't have a good SSR test so please try it out after it merges?